client/rpc_sender.go

```diff
@@ -83,6 +84,13 @@ func newRPCSender(server string, context *base.Context, retryOpts retry.Options,
 // and been executed successfully. We retry here to eventually get through with
 // the same client command ID and be given the cached response.
 func (s *rpcSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
+	timeout := 5 * s.retryOpts.MaxBackoff
```

Comment:
> Do you think the handshake itself is just taking long enough that the tests are flaking?

Yes, that appears to be what's happening. I can reproduce the failures locally (every time!) by running a race build under `stress -p 8`. If I change it to panic on timeout instead of returning an error I see a thread in `rsa.PrivateKey.Sign` in the TLS handshake. So it looks like we need to either switch back to smaller keys for (most) tests or increase the timeouts.